cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/common/AbstractWatcherThread.java

```diff
@@ -47,62 +66,62 @@
   private static final String DELETED = "DELETED";
   // The response type from K8s when a resource was modified
   private static final String MODIFIED = "MODIFIED";
+  // The response type from K8s when there is error from the watch
+  private static final String ERROR = "ERROR";
+
+  protected final String group;
+  protected final String version;
+  protected final String plural;
 
   private final String namespace;
   private final Random random;
   private final Type resourceType;
-  private volatile Watchable<T> watch;
+  private final AtomicReference<Watchable<DynamicKubernetesObject>> watch;
+  private final CachingResourceChangeListener<T> changeListener;
   private volatile boolean stopped;
+  private volatile ApiClient apiClient;
+
+  protected AbstractWatcherThread(String threadName, String namespace, String group, String version, String plural) {
+    this(threadName, namespace, group, version, plural, null);
+  }
 
-  protected AbstractWatcherThread(String threadName, String namespace) {
+  @VisibleForTesting
+  AbstractWatcherThread(String threadName, String namespace, String group, String version, String plural,
+                        @Nullable ApiClient apiClient) {
     super(threadName);
+    setDaemon(true);
```

Comment:
Are we moving the setDaemon() call from child classes to the base class? In some cases, looks like the child class calls setDaemon(true) (ex - in AppResourceWatcherThread https://github.com/cdapio/cdap/pull/14408/files#diff-c493972827be46b0a7ada1ec5226134f35a25de136d2c9282a06b980b3e10d9fL78). But in some other cases it doesn't seem to be called (ex - PodWatcherThread https://github.com/cdapio/cdap/blob/bc69c0a7999a32f304b77855a32db1b04e0d24e0/cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java#L1261)

Reply:
Yes we are. Just make all these threads daemon threads by default.